platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/AbstractMigrationTest.java

```diff
@@ -0,0 +1,387 @@
+     * @throws IOException if an I/O error occurs while reading the stream
+     */
+    public static Map<String, ZipEntry> getEntriesFrom(ByteArrayOutputStream baos)
+            throws IOException {
+        baos.close(); // not really required!
+        return AbstractMigrationTest.getEntriesFrom(new ByteArrayInputStream(baos.toByteArray()));
+    }
+
+    /**
+     * Retrieves all zip entries representing files from the provided input stream.
+     *
+     * @param in the input stream from which to retrieved all zip entries (will be closed)
+     * @return a map keyed by entry names with the corresponding entry
+     * @throws IOException if an I/O error occurs while reading the stream
+     */
+    public static Map<String, ZipEntry> getEntriesFrom(InputStream in) throws IOException {
+        try (final ZipInputStream zin = (in instanceof ZipInputStream) ?
+                (ZipInputStream) in :
+                new ZipInputStream(in)) {
+            final Map<String, ZipEntry> entries = new HashMap<>();
+
+            while (true) {
+                final java.util.zip.ZipEntry ze = zin.getNextEntry();
+
+                if (ze == null) {
+                    return entries;
+                } else if (!ze.isDirectory()) {
+                    entries.put(ze.getName(), new ZipEntry(ze, IOUtils.toByteArray(zin)));
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates test files with the given names in the specified directory resolved under ${ddf.home}.
+     * <p>
+     * <i>Note:</i> Each files will be created with the filename (no directory) as its content.
+     *
+     * @param dir   the directory where to create the test files
+     * @param names the names of all test files to create in the specified directory
+     * @return a list of all relativized paths from ${ddf.home} for all test files created
+     * @throws IOException if an I/O error occurs while creating the test files
+     */
+    public List<Path> createFiles(Path dir, String... names) throws IOException {
+        return createFiles(new ArrayList<>(), dir, names);
+    }
+
+    /**
+     * Creates test files with the given names in the specified directory resolved under ${ddf.home}
+     * and adds their corresponding relativized from ${ddf.home} paths to the given list.
+     * <p>
+     * <i>Note:</i> Each files will be created with the filename (no directory) as its content.
+     *
+     * @param paths a list of paths where to add all paths for the test files created
+     * @param dir   the directory where to create the test files
+     * @param names the names of all test files to create in the specified directory
+     * @return <code>paths</code> for chaining
+     * @throws IOException if an I/O error occurs while creating the test files
+     */
+    public List<Path> createFiles(List<Path> paths, Path dir, String... names) throws IOException {
+        final File rdir = DDF_HOME.resolve(dir)
+                .toFile();
+
```

Comment:
Can we add something to this method that creates all parent directories contained in the given path?

maybe just a call to `rdir.mkdirs()`